### PR TITLE
added cancel_replace_order, cancel_all_open_orders and cancel_all_open_margin_orders to client.py

### PR DESCRIPTION
**Pull request by m-HD**
- added cancel_replace_order()
- added cancel_all_open_orders()
- added cancel_all_open_margin_orders()
- enum TIME_IN_FORCE_GTD (good till date)
- parameters for get_current_order_count to set recvWindow

### CODE REVIEW

**Comment by carlosmiei** [comment]
Hello @m-HD,

Thanks a lot for your contribution. We will review and merge it soon.

File: binance/client.py [hunk 1]
```diff
@@ -2368,7 +2477,7 @@ def get_current_order_count(self):
             ]
 
         """
-        return self._get("rateLimit/order", True)
+        return self._get('rateLimit/order', True, data=params)
```
Comment (carlosmiei) [inline binance/client.py:2480]: this endpoint does not accept any parameter, so this is not needed

https://binance-docs.github.io/apidocs/spot/en/#query-unfilled-order-count-user_data

Comment (m-HD) [reply binance/client.py]: Binance Documentation says there is a recvWindow:
Old documentation: https://binance-docs.github.io/apidocs/spot/en/#query-unfilled-order-count-user_data
New documentation: https://developers.binance.com/docs/binance-spot-api-docs/rest-api/public-api-endpoints#query-unfilled-order-count-user_data

**Comment by m-HD** [comment]
I didn't write tests for the new methods.

**Comment by m-HD** [comment]
Anything I need to do?

**Comment by carlosmiei** [comment]
@m-HD don't worry all good 😅 (will just add a test before merging)

File: binance/client.py (hunk 1 above)
Comment (carlosmiei) [reply binance/client.py]: @m-HD Yes, but that's injected automatically. Nonetheless, I think we can keep params to keep the code consistent so don't worry

File: binance/client.py [hunk 2]
```diff
@@ -2160,6 +2160,115 @@ def cancel_order(self, **params):
...
+                }
+            }
+        Similar to POST /api/v3/order, additional mandatory parameters are determined by type.
+
+        Response format varies depending on whether the processing of the message succeeded, partially succeeded, or failed. 
+
+        :raises: BinanceRequestException, BinanceAPIException
+
+        """
+        return self._post("order/cancelReplace", True, data=params)
```
Comment (carlosmiei) [inline binance/client.py:2270]: I think we're missing overriding the version here as well

**Pull request merged**